### PR TITLE
build(bazel): Add Bazel builders

### PR DESCRIPTION
**Pull request by kyliau**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/bazel/src/builders/bazel.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/// <reference types='node'/>
+import {spawn, spawnSync} from 'child_process';
+import {Observable, Subject} from 'rxjs';
+
+export function runBazel(
+    projectDir: string, command: string, watch: boolean, workspaceTarget: string,
+    flags: string[]): Observable<void> {
+  const doneSubject = new Subject<void>();
+  const buildProcess = spawn(watch ? 'ibazel' : 'bazel', [command, workspaceTarget, ...flags], {
```
Comment (DevVersion) [inline packages/bazel/src/builders/bazel.ts]: I think there should be some better exception if `ibazel` is not available. We have the check for `bazel`, but that does not _always_ mean that `ibazel` is present.

Comment (kyliau) [reply packages/bazel/src/builders/bazel.ts]: Added logic to check both ibazel and bazel installation

File: packages/bazel/src/builders/bazel.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/// <reference types='node'/>
+import {spawn, spawnSync} from 'child_process';
+import {Observable, Subject} from 'rxjs';
+
+export function runBazel(
+    projectDir: string, executable: string, command: string, workspaceTarget: string,
```
Comment (alexeagle) [inline packages/bazel/src/builders/bazel.ts]: maybe command should have type `"build"|"test"|"run"|"coverage"|"query"`

File: packages/bazel/src/builders/bazel.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    flags: string[]): Observable<void> {
+  const doneSubject = new Subject<void>();
+  const buildProcess = spawn(executable, [command, workspaceTarget, ...flags], {
+    cwd: projectDir,
+    stdio: 'inherit',
+    shell: false,
+  });
+
+  buildProcess.once('close', (code: number) => {
+    code === 0 ? doneSubject.next() : doneSubject.error('Bazel failed.');
```
Comment (alexeagle) [inline packages/bazel/src/builders/bazel.ts]: should we propagate the exit code to the `ng` command?

File: packages/bazel/src/builders/bazel.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  return doneSubject.asObservable();
+}
+
+export function checkInstallation(executable: string, projectDir: string) {
+  const child = spawnSync(executable, ['version'], {
+    cwd: projectDir,
+    shell: true,
```
Comment (alexeagle) [inline packages/bazel/src/builders/bazel.ts]: shell is true here but false above, should they be the same?

File: packages/bazel/src/builders/builders.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "builders": {
+    "build": {
+      "class": "./index#BundleBuilder",
```
Comment (alexeagle) [inline packages/bazel/src/builders/builders.json]: throughout: the word `bundle` seems like it's leftover from when the builder was more specific. Remove?

File: packages/bazel/src/builders/builders.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+      "schema": "./schema.json",
+      "description": "Builds the Bazel bundle target."
```
Comment (alexeagle) [inline packages/bazel/src/builders/builders.json]: Builds -> executes bazel on a target?

File: packages/bazel/src/builders/bazel.ts (hunk 4 above)
Comment (kyliau) [reply packages/bazel/src/builders/bazel.ts]: `shell` is not needed here

**Review by alexeagle — APPROVED** [review]

**Pull request closed without merging**